osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs

```diff
@@ -209,7 +209,45 @@ public void TestStartAfterUnMatchingFilterDoesNotStart()
             AddAssert("start not requested", () => !startRequested);
         }
 
-        private void importForRuleset(int id) => AddStep($"import test map for ruleset {id}", () => manager.Import(createTestBeatmapSet(getImportId(), rulesets.AvailableRulesets.Where(r => r.ID == id).ToArray())));
+        [Test]
+        public void TestAddNewBeatmapWhileSelectingRandom()
+        {
+            const int test_count = 10;
+            int beatmapChangedCount = 0;
+            createSongSelect();
+            AddStep("Setup counter", () =>
+            {
+                beatmapChangedCount = 0;
+                songSelect.Carousel.BeatmapSetsChanged += () => beatmapChangedCount++;
+            });
+            AddRepeatStep($"Create beatmaps {test_count} times", () =>
+            {
+                manager.Import(createTestBeatmapSet(getImportId(), rulesets.AvailableRulesets.Where(r => r.ID == 0).ToArray()));
+
+                Scheduler.AddDelayed(() =>
+                {
+                    // Wait for debounce
+                    songSelect.Carousel.SelectNextRandom();
+                }, 400);
+            }, test_count);
+
+            AddAssert($"Beatmap changed {test_count} times", () => beatmapChangedCount == test_count);
+        }
+
+        [Test]
+        public void TestHideSetSelectsCorrectBeatmap()
+        {
+            int? previousID = null;
+            createSongSelect();
+            importForRuleset(0);
+            AddStep("Move to last difficulty", () => songSelect.Carousel.SelectBeatmap(songSelect.Carousel.BeatmapSets.First().Beatmaps.Last()));
+            AddStep("Store current ID", () => previousID = songSelect.Carousel.SelectedBeatmap.ID);
+            AddStep("Hide first beatmap", () => manager.Hide(songSelect.Carousel.SelectedBeatmapSet.Beatmaps.First()));
+            AddAssert("Selected beatmap has not changed", () => songSelect.Carousel.SelectedBeatmap.ID == previousID);
+        }
+
+        private void importForRuleset(int id) => AddStep($"import test map for ruleset {id}",
```

Comment:
This is duplicate code. Why was the original call site not replaced with a call to this?